pkg/segment/aggregations/segaggs.go

```diff
@@ -1482,6 +1487,124 @@ func performSortColRequestOnMeasureResults(nodeResult *structs.NodeResult, letCo
 	return nil
 }
 
+func performMultiValueColRequest(nodeResult *structs.NodeResult, letColReq *structs.LetColumnsRequest, recs map[string]map[string]interface{}) error {
+	if recs != nil {
+		if err := performMultiValueColRequestWithoutGroupby(letColReq, recs); err != nil {
+			return fmt.Errorf("performMultiValueColRequest: %v", err)
+		}
+		return nil
+
+	}
+
+	if err := performMultiValueColRequestOnHistogram(nodeResult, letColReq); err != nil {
+		return fmt.Errorf("performMultiValueColRequest: %v", err)
+
+	}
+
+	return nil
+}
+
+func performMultiValueColRequestWithoutGroupby(letColReq *structs.LetColumnsRequest, recs map[string]map[string]interface{}) error {
+	mvColReq := letColReq.MultiValueColRequest
+
+	for _, rec := range recs {
+		fieldValue, ok := rec[mvColReq.ColName]
+		if !ok {
+			continue
+		}
+
+		fieldValueStr, ok := fieldValue.(string)
+		if !ok {
+			fieldValueStr = fmt.Sprintf("%v", fieldValue) // Convert to string
+		}
+
+		switch mvColReq.Command {
+		case "makemv":
+			finalValue := performMakeMV(fieldValueStr, mvColReq)
+			rec[mvColReq.ColName] = finalValue
+		default:
+			return fmt.Errorf("performMultiValueColRequestWithoutGroupby: unknown command %s", mvColReq.Command)
+		}
+	}
+
+	return nil
+}
+
+func performMultiValueColRequestOnHistogram(nodeResult *structs.NodeResult, letColReq *structs.LetColumnsRequest) error {
+	// mvColReq := letColReq.MultiValueColRequest
+
+	for _, aggregationResult := range nodeResult.Histogram {
+		for _, bucketResult := range aggregationResult.Results {
+			fieldValue, index, isStatRes := bucketResult.GetBucketValueForGivenField(letColReq.MultiValueColRequest.ColName)
+			if fieldValue == nil {
+				continue
+			}
+
+			if isStatRes {
+				return fmt.Errorf("performMultiValueColRequestOnHistogram: field %s is a statistic result. Cannot perform Multi value string operations on a Statistic result", letColReq.MultiValueColRequest.ColName)
+			}
+
+			fieldValueStr, ok := fieldValue.(string)
+			if !ok {
+				fieldValueStr = fmt.Sprintf("%v", fieldValue) // Convert to string
+			}
+
+			switch letColReq.MultiValueColRequest.Command {
+			case "makemv":
+				finalValue := performMakeMV(fieldValueStr, letColReq.MultiValueColRequest)
+				err := bucketResult.SetBucketValueForGivenField(letColReq.MultiValueColRequest.ColName, finalValue, index, isStatRes)
+				if err != nil {
+					log.Errorf("performMultiValueColRequestOnHistogram: error setting bucket value for field %s: %v", letColReq.MultiValueColRequest.ColName, err)
+					continue
+				}
+			default:
+				return fmt.Errorf("performMultiValueColRequestOnHistogram: unknown command %s", letColReq.MultiValueColRequest.Command)
+			}
+		}
+	}
+
+	return nil
+
+}
+
+func performMakeMV(strVal string, mvColReq *structs.MultiValueColLetRequest) interface{} {
+	if strVal == "" {
+		return ""
+	}
+
+	var values []string
+	if mvColReq.IsRegex {
+		re := regexp.MustCompile(mvColReq.DelimiterString)
```

Comment:
I think `DelimiterString` is provided by the user. If it is, you shouldn't use `MustCompile()` because bad user input will cause a panic. Instead use `Compile()` and check for an error

Reply:
In the parser, while parsing the regex, I am checking with `regexp.Complie` and it will throw an error if regex is not valid. So I believe this should be fine right? As we are sure it will only come here, if the regex is valid.

<img width="889" alt="image" src="https://github.com/siglens/siglens/assets/29257812/01d73d3c-66c5-48e4-8957-5caed8009b34">